Source/JavaScriptCore/runtime/JSGlobalObject.h

```diff
@@ -1077,23 +1079,30 @@ class JSGlobalObject : public JSSegmentedVariableObject {
     static void reportViolationForUnsafeEval(const JSGlobalObject*, const String&) { }
 
     bool evalEnabled() const { return m_evalEnabled; }
+    bool trustedEvalEnabled() const { return m_trustedEvalEnabled; }
     bool webAssemblyEnabled() const { return m_webAssemblyEnabled; }
     bool requiresTrustedTypes() const { return m_requiresTrustedTypes; }
+    bool requiresTrustedTypesEnforced() const { return m_requiresTrustedTypesEnforced; }
     const String& evalDisabledErrorMessage() const { return m_evalDisabledErrorMessage; }
     const String& webAssemblyDisabledErrorMessage() const { return m_webAssemblyDisabledErrorMessage; }
     void setEvalEnabled(bool enabled, const String& errorMessage = String())
     {
         m_evalEnabled = enabled;
         m_evalDisabledErrorMessage = errorMessage;
     }
+    void setTrustedEvalEnabled(bool enabled)
+    {
+        m_trustedEvalEnabled = enabled;
+    }
     void setWebAssemblyEnabled(bool enabled, const String& errorMessage = String())
     {
         m_webAssemblyEnabled = enabled;
         m_webAssemblyDisabledErrorMessage = errorMessage;
     }
-    void setRequiresTrustedTypes(bool required)
+    void setRequiresTrustedTypes(bool required, bool enforced)
     {
         m_requiresTrustedTypes = required;
+        m_requiresTrustedTypesEnforced = enforced;
     }
```

Comment:
The old version of this function was easy to understand at the call site. The new one is not. It’s not at all clear what `false, false` means. Can we come up with something where it’s easier to read and understand use of this? Two separate functions would be easier to understand, like `setEnforcesTrustedTypesRequirement`, or could pass in an enumeration with three values for _not required_, _required but not enforced_, _required and enforced_.